### ISSUE TITLE
Running --interactive mode still modifies my libs.versions.toml file

### ISSUE THREAD

**Issue by Zordid** [littlerobots/version-catalog-update-plugin#146]
**Describe the issue**
As per documentation, running in interactive mode should *not* modify the libs.versions.toml file - but it does.
It changes how my plugins are declared, from `{ id = "...", version = "..." }` to plain string declaration, it removes blank lines, it removes seemingly unused versions and libraries, it changes how I arrange my bundles, etc.

**Steps to reproduce**
Just run interactive mode and see that your file is changed.
The expected interactive file is also created...

**Comment by hvisser** [comment]
Yes, currently it will still format the toml file. I think I did this to process unused libraries. The formatting is kind of inherent to the plugin and it would also be applied when adding the updates (basically every time the file is rewritten).

I'll take a look to see if the format pass can be removed when running interactive though. It might need waiting until the plugin has switched to the internal resolver for that. After switching there's no detection of unused dependencies so it's possible that the current behaviour isn't needed anymore.

**Comment by polyoxidonium** [comment]
@hvisser 
This is still an issue. 
I wanted to use just interactive mode because the original task mess up with my formatting but now I have to revert changes on the toml file.
And btw it does not affect only formatting, but it also removes "unused" dependencies in a false-positive way (boms that are not explicitly referenced) unless the configuration is not adapted.

**Comment by hvisser** [comment]
I'm aware, that's why this issue is still open. What I said in https://github.com/littlerobots/version-catalog-update-plugin/issues/146#issuecomment-1978288525 still holds; the next version (1.0) will switch over to the new resolver, which will also fix "disappearing" dependencies that you are experiencing. The new resolver is behind a flag currently, see the release notes for details.
